vizro-core/src/vizro/models/_navigation/nav_bar.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import annotations
+
+import itertools
+from typing import List, Optional
+
+from dash import html
+from pydantic import validator
+
+from vizro.managers import model_manager
+from vizro.models import VizroBaseModel
+from vizro.models._models_utils import _log_call
+from vizro.models._navigation.nav_item import NavItem
+from vizro.models.types import NavPagesType
+
+
+class NavBar(VizroBaseModel):
+    """Navigation bar to be used as a selector for `Navigation`.
+
+    Args:
+        pages (Optional[NavPagesType]): See [`NavPagesType`][vizro.models.types.NavPagesType].
+            Defaults to `None`.
+        items (List[NavItem]): List of NavItem models. Defaults to `[]`.
+    """
+
+    pages: Optional[NavPagesType] = None
+    items: List[NavItem] = []
+
+    @_log_call
+    def pre_build(self):
+        from vizro.models._navigation import Navigation
+
+        _, navigation = next(model_manager._items_with_type(Navigation))
+
+        if self.pages is None:
+            self.pages = navigation.pages
```

Comment:
I don't quite understand why this is needed here. Could you elaborate on that?

In my mind, it's supposed to work like this:
- `NavBar.pages` is obligatory similar to `Accordion.pages`
- If `NavBar.pages` or `Accordion.pages` are not defined, they are auto-populated from `Navigation.pages` (this logic should probably live in the pre-build of Navigation - similar how we do auto.population for the selectors in Filter)
- If `NavBar.pages` or `Accordion.pages` are defined, they take precedence over `Navigation.pages`

Am I missing something or did you hit any blockers that didn't enable above? It's very likely that I am just missing someting 😄